### PR TITLE
Multiple plans dropdown

### PR DESCRIPTION
**Pull request by elizabeth-tang**
### Summary 

- Implemented frontend for multiple plans dropdown
- Created cypress tests for open/close states

This is the second front end task to implement the Multiple Plans feature. 

### Test Plan 

- In the console, enter GK.enableMultiplePlans()
- Is dropdown initially closed?
- Does it open when clicked. Does it close when clicked again? Does arrow flip? 
- Does "No additional plans" show when only one plan is in the data?
- Does current plan text change when another plan is clicked?

### CODE REVIEW

**Review by mirandayu131 — CHANGES_REQUESTED** [review]
Can you include in your test plan how to enable the feature flag so I can see your changes?

File: cypress/integration/test.spec.ts [hunk 1]
```diff
@@ -237,3 +238,15 @@ it('Minimize a semester', () => {
   cy.get('[data-cyId=semester-course]').should('be.visible');
   cy.get('[data-cyId=semester-addCourse]').should('be.visible');
 });
+
+it('Multiple plans dropdown open/close', () => {
+  window.GK.enableMultiplePlans();
+  // dropdown initially closed
+  cy.get('[data-cyId=multiplePlans-wrapper]').should('not.have.class', 'shown');
```
Comment (zachary0kent) [inline cypress/integration/test.spec.ts:245]: Would `.should('be.visible')` work here instead?

File: package.json [hunk 2]
```diff
@@ -57,7 +57,7 @@
     "autoprefixer": "10.4.5",
     "babel-jest": "^26.6.3",
     "concurrently": "^6.0.2",
-    "cypress": "^6.6.0",
+    "cypress": "^6.9.1",
```
Comment (zachary0kent) [inline package.json:60]: Any reason we bumped cypress?

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        :class="{ plansHidden: shown }"
+      >
+        <img :src="editplan" class="multiplePlans-dropdown-placeholder editimg" />
+        <div class="multiplePlans-dropdown-placeholder plan">{{ currplan }}</div>
+        <div class="multiplePlans-dropdown-placeholder down-arrow" v-if="!shown"></div>
+        <div class="multiplePlans-dropdown-placeholder up-arrow" v-if="shown"></div>
+      </div>
+      <div class="multiplePlans-dropdown-content" v-if="shown">
+        <div
+          v-for="plan in filterPlans"
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:17]: Can you rename the `plan` identifier used in the `v-for`?

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    return {
+      plans: ['PLAN 1', 'PLAN 2'],
+      shown: false,
+      currplan: 'PLAN 1',
+      editplan,
+    };
+  },
+  computed: {
+    filterPlans() {
+      const filtered = this.$data.plans.filter(plan => plan !== this.$data.currplan);
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:47]:
```suggestion
      const filtered = this.plans.filter(plan => plan !== this.currplan);
```

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+export default defineComponent({
+  props: {
+    plan: { type: String, default: 'No additional plans yet' },
+  },
+  data() {
+    return {
+      plans: ['PLAN 1', 'PLAN 2'],
+      shown: false,
+      currplan: 'PLAN 1',
+      editplan,
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:42]: Can we use camelCase?

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+      <div
+        class="multiplePlans-dropdown-placeholder wrapper"
+        @click="closeDropdownIfOpen()"
+        data-cyid="multiplePlans-wrapper"
+        :class="{ plansHidden: shown }"
+      >
+        <img :src="editplan" class="multiplePlans-dropdown-placeholder editimg" />
+        <div class="multiplePlans-dropdown-placeholder plan">{{ currplan }}</div>
+        <div class="multiplePlans-dropdown-placeholder down-arrow" v-if="!shown"></div>
+        <div class="multiplePlans-dropdown-placeholder up-arrow" v-if="shown"></div>
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:13]:
```suggestion
        <div class="multiplePlans-dropdown-placeholder up-arrow" v-if="shown"></div>
        <div class="multiplePlans-dropdown-placeholder down-arrow" v-else></div>
```

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  data() {
+    return {
+      plans: ['PLAN 1', 'PLAN 2'],
+      shown: false,
+      currplan: 'PLAN 1',
+      editplan,
+    };
+  },
+  computed: {
+    filterPlans() {
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:46]: Can you rename this to something more descriptive like `otherPlans` or `deselectedPlans`?

File: src/components/Requirements/RequirementSideBar.vue [hunk 8]
```diff
@@ -38,7 +38,11 @@
           </button>
           <div class="multiple-plans" v-if="multiplePlansAllowed">
             <button class="add-plan-button" @click="addPlan()">+ Add Plan</button>
+            <div class="multiple-plans-dropdown">
+              <multiple-plans-dropdown></multiple-plans-dropdown>
```
Comment (zachary0kent) [inline src/components/Requirements/RequirementSideBar.vue:42]:
```suggestion
              <multiple-plans-dropdown />
```

**Review by zachary0kent — CHANGES_REQUESTED** [review]
Screenshot 2023-03-26 at 9 56 46 PM
Looks good overall! I just encountered one issue when shrinking the size of the screen; it seems the padding between the arrow and the right side of the dropdown changes as you adjust the screen size.

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+  &-placeholder {
+    height: 100%;
+    font-size: 16px;
+    font-weight: bold;
+    line-height: 16px;
+    margin-left: 0.25rem;
+    display: flex;
+    align-items: center;
+
+    color: #636363;
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:76]: Can we either use a similar color from `src/assets/constants/colors` or add this there if need be?

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+  &-content {
+    margin-left: 0.11rem;
+    z-index: 100;
+    max-height: 8rem;
+    border-radius: 2px;
+    &.item {
+      width: 91%;
+      border: 1px solid #636363;
+      z-index: 150;
+      background-color: #ffffff;
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:129]:
```suggestion
      background-color: white;
```

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+      height: 2.5rem;
+      font-style: normal;
+      font-weight: normal;
+      font-size: 16px;
+      line-height: 16px;
+      display: flex;
+      align-items: center;
+      color: #858585;
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:137]: Same comment about this color

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+    max-height: 8rem;
+    border-radius: 2px;
+    &.item {
+      width: 91%;
+      border: 1px solid #636363;
+      z-index: 150;
+      background-color: #ffffff;
+      height: 2.5rem;
+      font-style: normal;
+      font-weight: normal;
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:132]: What happens if these properties aren't set?

**Review by zachary0kent — COMMENTED** [review]
Also, can we avoid use of pixels? I see you're using rem in some places, which is good.

File: src/components/Requirements/MultiplePlansDropdown.vue [hunk 13]
```diff
@@ -0,0 +1,142 @@
+<template>
+  <div>
+    <div class="multiplePlans-dropdown">
+      <div class="multiplePlans-dropdown-placeholder wrapper" @click="closeDropdownIfOpen()">
+        <img :src="editPlan" class="multiplePlans-dropdown-placeholder editimg" />
+        <div class="multiplePlans-dropdown-placeholder plan">{{ currPlan }}</div>
+        <div class="multiplePlans-dropdown-placeholder up-arrow" v-if="shown"></div>
+        <div class="multiplePlans-dropdown-placeholder down-arrow" v-else></div>
```
Comment (zachary0kent) [inline src/components/Requirements/MultiplePlansDropdown.vue:8]: Is there a reason you didn't use `img` tags for the arrow?

**Review by zachary0kent — CHANGES_REQUESTED** [review]
Screenshot 2023-03-28 at 4 06 29 PM
Can you add a hover state for hovering over a different plan? Also, can you add more plans just to make sure they align correctly in the dropdown?

**Review by zachary0kent — APPROVED** [review]
Looks good to me!

**Review by zachary0kent — CHANGES_REQUESTED** [review]
Actually, sorry for one last request, but if you copied the styling from another dropdown, can you factor the CSS into an SCSS file?

**Comment by zachary0kent** [comment]
Thanks for factoring out the styling! Can you import the stylesheet from where you had copied the styling?

**Review by zachary0kent — APPROVED** [review]
Looks great, thanks for the hard work!

**Review by mirandayu131 — APPROVED** [review]
Looks good to me :)

**Pull request merged**